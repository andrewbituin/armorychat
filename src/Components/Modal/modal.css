
.modal {
  position: relative;
  font-family: Rajdhani, sans-serif;
  color: white;
  background-color: #2c2e34;
  border: 1px solid #6e6f72;
  border-radius: 5px;
  z-index: 7;
  box-shadow: 0 5px 13px 0 #00000080;
}

.modal-wrapper:focus {
  outline: none;
}

.modal-header {
  background-image: linear-gradient(#363740, #2c2e34);
  margin: 0 -20px 10px;
  padding: 5px 0 10px 10px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  font-size: 18px;
}

.modal-overlay {
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #00000086;
  z-index: 6;
}

.modal-close-button {
  position: absolute;
  top: 5px;
  right: 5px;
  font-size: 18px !important;
  width: 30px;
  height: 30px;
}

@media screen and (max-width: 900px) {
  .modal {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    height: 100%;
    border: none;
    border-radius: 0;
    overflow-y: auto;
    overflow-x: hidden;
  }
}