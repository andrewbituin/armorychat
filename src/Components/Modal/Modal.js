import React, { useEffect } from 'react';
import ReactDOM from 'react-dom';
import './modal.css'

export default function Modal({ isShowing, hide, content, noDisableScrolling }) {
  return ReactDOM.createPortal(
    (isShowing)
      ? <ModalComponent content={content} hide={hide} noDisableScrolling={noDisableScrolling}/>
      : null,
    document.body
  );
}

function ModalComponent({ hide, content, noDisableScrolling }) {
  useEffect(() => {
    if (!noDisableScrolling) {
      document.body.classList.add('no-scroll');
    }
    
    return () => {
      if (!noDisableScrolling) {
        document.body.classList.remove('no-scroll');
      }
    }
  }, []);

  return (
    <div className="modal-overlay">
      <div className="modal-wrapper" aria-modal aria-hidden tabIndex={-1} role="dialog">
        <div className="modal">
          <button type="button" className="modal-close-button green-button" data-dismiss="modal" aria-label="Close" onClick={hide}>
            <i className="fas fa-times"/>
          </button>
          {content}
        </div>
      </div>
    </div>
  );
}