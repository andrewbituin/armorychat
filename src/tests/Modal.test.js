import React from 'react';
import Modal from '../Components/Modal/Modal';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

describe('Modal component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Modal />
      </MemoryRouter>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  })

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Modal />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})