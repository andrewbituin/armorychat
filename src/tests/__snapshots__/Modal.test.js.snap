// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders the UI as expected 1`] = `null`;
